### DISCUSSION TITLE
📑 Weekly report 09.11. - 15.11.23

### DISCUSSION THREAD

**Discussion by MargotGsb** [SE-TINF22B6/Plapy#82] [Weekly reports]
Hello and welcome everyone! We are glad to show you our latest progress ╰ (´꒳`) ╯
This week we made a simplyfied Class Diagramm and an ER Diagramm.

### Class Diagramm
The Class Diagramm we made, is a simplyfied version of our Intellij auto created Diagramms. Its purpose is to get a better overview of our project. If you want to see the full version Click on this link: [Class Diagrams](https://github.com/SE-TINF22B6/Plapy/tree/e851602f6067315c1be7dc20413b0861cef66710/Class%20Diagrams)

![Klassendiagramm drawio (3)](https://github.com/SE-TINF22B6/Plapy/assets/123726628/3d50e2cb-9957-499f-8f07-c62056121b10)

**Further Explaination of the Class Diagramm:**
- The Command Interface has more implementations than just the Play Command, but it would be too much for an overseeable Diagramm ![image](https://github.com/SE-TINF22B6/Plapy/assets/123726628/d6b63d1a-d776-4c23-aa6d-4dd6ee5b5d2c)
- The DiscordIndex Class is distributed by Discord. Through this Class our Bot can communicate with the Discord API ![image](https://github.com/SE-TINF22B6/Plapy/assets/123726628/344d6beb-67fa-4cdd-b5fc-ec43f111310d)
- The bot Class is sort of the head of our programm. It gets the ChatInput from the DiscordAPI and then proceeds to initiate the right step for each input it gets. (Example: User typed "/play *some Song-URL from youtube*". The bot will give the Song to the MusicQueue and there it will be added to the current Queue)
![image](https://github.com/SE-TINF22B6/Plapy/assets/123726628/39515891-3603-48fe-accf-c4e09e67122d)

### ER Diagramm
Our ER Diagramm is fairly simple. We only have Songs and Playlists to save.

![Plapy-ER drawio (3)](https://github.com/SE-TINF22B6/Plapy/assets/123726628/60e53b06-1bb5-4bb7-824c-8f94e6bf50ea)

**Further Explaination of the ER Diagramm:**

Relationships: A Playlist can feature n Songs. And a Song can be fearured in m Playlists. So it would be a n to m Relationship. But we resolved the n to m Relationship with the connecting Enity "features".

**Added Features this week**

The radio command has recieved two notable improvements:
- Duration filter
- Duplicates filter
Both filters are active by default and can be disabled when executing the command. The duration filter disregards all songs that are longer than 15min to avoid filling the queue with mixes. The duplicates filter uses string manipulation to extract the song title out of a video title and then checks all songs added after it for the string to avoid adding multiple versions of the same song if it is uploaded more than one time or in more than one version to YouTube

I hope you found this report to be informative or interesting.
Until Next Week!
- Your Team Plapy

**Comment by derrrobin** [comment]
Hello Team Plapy,

Thank you for sharing your latest progress! Your simplified Class Diagram provides a clear overview of your project, and I appreciate the effort to enhance understanding with a link to the full version. The explanation of the Command Interface, DiscordIndex Class, and the role of the bot in handling DiscordAPI inputs is insightful, making it easier for readers to grasp the architecture.

The ER Diagram for Songs and Playlists is well-organized, and the explanation of the n to m relationship resolution using the "features" entity adds clarity to your data model.

Your CookPit Team
